package-builder/flox-build.mk

```diff
@@ -218,8 +219,25 @@ endef
 # to the inner "build wrapper" environment in the in-situ build mode in support
 # of the tools and compilers found in the outer environment, and for Flox to
 # otherwise function properly.
-ALLOW_OUTER_ENV_VARS := FLOX_RUNTIME_DIR HOME PATH \
-  $(filter NIX_CFLAGS% NIX_CC%,$(.VARIABLES))
+#
+# Vars required for flox functionality.
+ALLOW_OUTER_ENV_VARS := \
+  FLOX_ACTIVATE_TRACE FLOX_ENV_DIRS FLOX_RUNTIME_DIR HOME PATH
+# Vars exported by 0100_common-paths.sh. Not included:
+# - LIBRARY_PATH : embedded in runpath of executables
+ALLOW_OUTER_ENV_VARS += \
+  INFOPATH CPATH PKG_CONFIG_PATH ACLOCAL_PATH XDG_DATA_DIRS \
+  LD_AUDIT GLIBC_TUNABLES DYLD_FALLBACK_LIBRARY_PATH
+# Vars exported by 0500_python.sh.
+ALLOW_OUTER_ENV_VARS += PYTHONPATH PIP_CONFIG_FILE
+# Vars exported by 0501_rust.sh.
+ALLOW_OUTER_ENV_VARS += RUST_SRC_PATH
+# Vars exported by 0502_jupyter.sh.
+ALLOW_OUTER_ENV_VARS += JUPYTER_PATH
+# Vars exported by 0800_cuda.sh.
+ALLOW_OUTER_ENV_VARS += LD_FLOXLIB_FILES_PATH
+# Vars set by Nix stdenv hooks.
+ALLOW_OUTER_ENV_VARS += $(filter NIX_CFLAGS% NIX_CC%,$(.VARIABLES))
```

Comment:
nonblocking note: I think it may be hard to maintain all these variables going forward